filebeat/crawler/registrar.go

```diff
@@ -16,7 +17,8 @@ type Registrar struct {
 	// Path to the Registry File
 	registryFile string
 	// Map with all file paths inside and the corresponding state
-	State map[string]*FileState
+	state      map[string]*FileState
+	stateMutex *sync.Mutex
```

Comment:
why pointer to mutex? Zero initialized sync.Mutex is already value. Advantage/Drawback of plain `sync.Mutex` is, it can not be copied (`go vet` will complain)